torch/nn/modules/module.py

```diff
@@ -997,8 +997,45 @@ def _load_from_state_dict(self, state_dict, prefix, local_metadata, strict,
                     if input_name not in self._modules and input_name not in local_state:
                         unexpected_keys.append(key)
 
+    def _strip_prefix_from_state_dict_if_exists(self, state_dict, prefix: str):
+        r"""This function strips a prefix if it exists in the model
+        state_dict and metadata.
+
+        Arguments:
+            state_dict (dict) : DP/DDP pytorch model state_dict
+            prefix (str) : Prefix to be removed from the DP/DDP model state_dict to
+                           to make it compatible with regular pytorch model.
+        Returns:
+            state_dict (dict)
+        """
+        keys = sorted(state_dict.keys())
+        if not all(len(key) == 0 or key.startswith(prefix) for key in keys):
+            return
+
+        for key in list(state_dict.keys()):
+            new_key = key[len(prefix):]
+            state_dict[new_key] = state_dict.pop(key)
+
+        try:
+            metadata = state_dict._metadata
+        except AttributeError:
+            pass
```

Comment:
Why do we ignore this exception? This would mean metadata could be `None` below resulting in an exception thrown from there which would be confusing.